hoomd/md/minimize/fire.py

```diff
@@ -0,0 +1,198 @@
+        while not(fire.has_converged()):
+           sim.run(100)
+
+    Note:
+        The algorithm requires a base integrator to update the particle
+        position and velocities. Usually this will be either NVE (to minimize
+        energy) or NPH (to minimize energy and relax the box). The quantity
+        minimized is in any case the energy (not the enthalpy or any other
+        quantity).
+
+    Note:
+        As a default setting, the algorithm will start with a :math:`\\delta t
+        = \\frac{1}{10} \\delta t_{max}` and attempts at least 10 search steps.
+        In practice, it was found that this prevents the simulation from making
+        too aggressive a first step, but also from quitting before having found
+        a good search direction. The minimum number of attempts can be set by
+        the user.
+
+    """
+    _cpp_class_name = "FIREEnergyMinimizer"
+
+    def __init__(self,
+                 dt,
+                 aniso='auto',
+                 forces=None,
+                 constraints=None,
+                 methods=None,
+                 rigid=None):
+
+        super().__init__(forces, constraints, methods, rigid)
+
+        self._param_dict.update(
+            ParameterDict(
+                dt=float(dt),
+                aniso=OnlyFrom(['true', 'false', 'auto'],
+                               preprocess=_preprocess_aniso),
+                min_steps_adapt=OnlyTypes(int,
+                                          preprocess=_raise_if_nonpositive),
+                finc_dt=float,
+                fdec_dt=float,
+                alpha_start=float,
+                fdec_alpha=float,
+                force_tol=float,
+                angmom_tol=float,
+                energy_tol=float,
+                min_steps_conv=OnlyTypes(int, preprocess=_raise_if_nonpositive),
+                _defaults={
+                    "aniso": "auto",
+                    "min_steps_adapt": 5,
+                    "finc_dt": 1.1,
+                    "fdec_dt": 0.5,
+                    "alpha_start": 0.1,
+                    "fdec_alpha": 0.99,
+                    "force_tol": 0.1,
+                    "angmom_tol": 0.1,
+                    "energy_tol": 1e-5,
+                    "min_steps_conv": 10
+                }))
+
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            cls = getattr(_md, self._cpp_class_name)
+        else:
+            cls = getattr(_md, self._cpp_class_name + "GPU")
+        self._cpp_obj = cls(self._simulation.state._cpp_sys_def, self.dt)
+        super()._attach()
+
+    def get_energy(self):
+        """Returns the energy after the last iteration of the minimizer."""
+        return self._cpp_obj.getEnergy()
+
+    def has_converged(self):
+        """Test if the energy minimizer has converged.
+
+        Returns:
+            True when the minimizer has converged. Otherwise, return False.
+        """
+        return self._cpp_obj.hasConverged()
```

Comment:
Make properties.

They should be loggable too, `converged` shouldn't be by default though.